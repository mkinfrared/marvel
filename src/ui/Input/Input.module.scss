@import "styles/colors";

.Input {

  label {
    display: block;
    margin-bottom: 5px;
  }

  input {
    background-color: rgba(255, 255, 255, 0.2);
    border: 1px solid $secondary-main;
    border-radius: 4px;
    height: 56px;
    outline: none;
    padding: 18px 14px;
    transition: border-color 250ms ease-in-out;
    width: 100%;

    &:hover {
      border-color: $primary-main;
    }

    &:focus {
      border-color: $primary-main;
      border-width: 2px;
    }
  }
}
