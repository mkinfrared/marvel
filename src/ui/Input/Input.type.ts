export interface InputProps {
  forwardRef?: React.Ref<HTMLInputElement>;
  label?: string;
  onBlur?: (event: React.FocusEvent<HTMLInputElement>) => void;
  onChange: (event: React.ChangeEvent<HTMLInputElement>) => void;
  onFocus?: (event: React.FocusEvent<HTMLInputElement>) => void;
  placeholder?: string;
  type?: "text" | "password";
  value: string;
}
