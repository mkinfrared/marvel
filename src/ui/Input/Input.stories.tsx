import { Meta, Story } from "@storybook/react/types-6-0";
import React from "react";

import { InputProps } from "ui/Input/Input.type";

import { Input } from "./Input";

export default {
  title: "UI/Input",
  component: Input
} as Meta;

const Template: Story<InputProps> = (args) => (
  <div style={{ width: 300 }}>
    <Input {...args} />
  </div>
);

const Primary = Template.bind({});

Primary.args = {
  label: "This is label"
  // onChange: { action: "Change" } as any,
  // onFocus: { action: "Focus" } as any,
  // onBlur: { action: "Blur" } as any
};

export { Primary };
