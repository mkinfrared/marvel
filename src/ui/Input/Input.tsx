import React, { useRef } from "react";

import assignForwardedRefs from "utils/assignForwardedRefs";

import css from "./Input.module.scss";
import { InputProps } from "./Input.type";

const Input = ({
  forwardRef,
  label,
  onBlur,
  onChange,
  onFocus,
  placeholder,
  value,
  type = "text"
}: InputProps) => {
  const innerRef = useRef<HTMLInputElement | null>(null);

  const setRefs = (ref: HTMLInputElement | null) => {
    if (forwardRef) {
      assignForwardedRefs(forwardRef, ref);
    }

    innerRef.current = ref;
  };

  const handleLabelClick = (event: React.SyntheticEvent<HTMLLabelElement>) => {
    event.preventDefault();

    innerRef.current?.focus();
  };

  return (
    <div className={css.Input}>
      {label && <label onClick={handleLabelClick}>{label}</label>}
      <input
        onChange={onChange}
        ref={setRefs}
        placeholder={placeholder}
        type={type}
        value={value}
        onBlur={onBlur}
        onFocus={onFocus}
      />
    </div>
  );
};

export { Input };

export default React.memo(Input);
