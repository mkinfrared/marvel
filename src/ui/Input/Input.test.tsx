import { render } from "@testing-library/react";
import React from "react";

import { Input } from "./Input";

describe("<Input />", () => {
  const Component = Input;
  const value = "marklar";
  const onChange = jest.fn();
  const props = { value, onChange };

  it("should be defined", () => {
    const { container } = render(<Component {...props} />);

    expect(container).toBeDefined();
  });

  it("match the snapshot", () => {
    const { container } = render(<Component {...props} />);

    expect(container).toMatchSnapshot();
  });
});
