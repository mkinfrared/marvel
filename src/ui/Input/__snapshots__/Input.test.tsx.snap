// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input /> match the snapshot 1`] = `
<div>
  <div
    class="Input"
  >
    <input
      type="text"
      value="marklar"
    />
  </div>
</div>
`;
